activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -476,6 +476,8 @@ def get_period_and_ensure_valid_local_time(period)
       end
 
       def transfer_time_values_to_utc_constructor(time)
+        # avoid creating another Time object if possible
+        return time if time.utc? && time.is_a?(Time)
```

Comment:
Shouldn't these two conditions be swapped? I'm assuming we wouldn't want to call `utc?` on an object that isn't a `Time` instance. Or am I misunderstanding the context? (very likely)